src/main/java/address/browser/HyperBrowser.java

```diff
@@ -0,0 +1,215 @@
+        initialiseHyperBrowser();
+    }
+
+    private void initialiseHyperBrowser(){
+        this.hyperBrowserView = new AnchorPane();
+
+        if (initialScreen.isPresent()) {
+            hyperBrowserView.getChildren().add(initialScreen.get());
+        }
+
+        pages = new ArrayList<>(noOfPages);
+        inActiveBrowserStack = new Stack<>();
+
+        for (int i=0; i<noOfPages; i++){
+            EmbeddedBrowser browser = new JxBrowserAdapter(new Browser());
+            FxViewUtil.applyAnchorBoundaryParameters(browser.getBrowserView(), 0.0, 0.0, 0.0, 0.0);
+            inActiveBrowserStack.push(browser);
+        }
+    }
+
+    public AnchorPane getHyperBrowserView() {
+        return hyperBrowserView;
+    }
+
+    public void dispose() {
+        inActiveBrowserStack.stream().forEach(embeddedBrowser -> embeddedBrowser.dispose());
+    }
+
+    /**
+     * Clears page from the paging system of HyperBrowser that are not required anymore.
+     * @param url The URL of page that is to be cleared, if exists.
+     * @return An optional page that is cleared from the pool of pages.
+     */
+    public synchronized void clearPage(URL url) {
+        Optional<Page> page = pages.stream().filter(p
+                -> {
+            try {
+                return UrlUtil.compareBaseUrls(url, p.getBrowser().getUrl());
+            } catch (MalformedURLException e) {
+                return false;
+            }
+        }).findAny();
+
+        if (page.isPresent()){
+            if (hyperBrowserView.getChildren().contains(page.get().getBrowser().getBrowserView())){
+                hyperBrowserView.getChildren().remove(0);
+            }
+            inActiveBrowserStack.push(page.get().getBrowser());
+            pages.remove(page.get());
+        }
+        assert pages.size() + inActiveBrowserStack.size() == NUMBER_OF_PRELOADED_PAGE;
+    }
+
+    /**
+     * Gets a list of URL from HyperBrowser displayed page and cached pages.
+     * @return A list of URL from HyperBrowser displayed page and cached pages.
+     */
+    public ArrayList<URL> getCachedPagesUrl(){
+        return pages.stream().map(page -> {
+            try {
+                return page.getBrowser().getUrl();
+            } catch (MalformedURLException e) {
+                return null;
+            }
+        }).collect(Collectors.toCollection(ArrayList::new));
+    }
+
+    /**
+     * Loads the URLs to the browser.
+     * @param url The URL of the content to load.
+     * @param futureUrl The URLs that may be called to load in the next T time.
+     * @return The page of the URL content.
+     * @throws IllegalArgumentSizeException When the amount of URLs to load is more than no of pages the paging system
+     *                                      of the HyperBrowser has.
+     * @throws NullPointerException When url is null.
+     */
+    public synchronized Page loadUrls(URL url, ArrayList<URL> futureUrl) throws IllegalArgumentSizeException
```

Comment:
The user should not now about this detail, the user just want to loadUrl to this browser